pkg/activator/net/throttler.go

```diff
@@ -246,29 +243,36 @@ func (rt *revisionThrottler) resetTrackers() {
 	}
 }
 
+// updateCapacity updates the capacity of the throttler and recomputes
+// the assigned trackers to the Activator instance.
+// Currently updateCapacity is ensured to be invoked from a single go routine
+// and this does not synchronize
 func (rt *revisionThrottler) updateCapacity(backendCount int) {
 	// We have to make assignments on each updateCapacity, since if number
 	// of activators changes, then we need to rebalance the assignedTrackers.
 
 	ac, ai := int(atomic.LoadInt32(&rt.numActivators)), int(atomic.LoadInt32(&rt.activatorIndex))
 	numTrackers := func() int {
-		rt.mux.Lock()
-		defer rt.mux.Unlock()
+		// We do not have to process the `podTrackers` under lock, since
+		// updateCapacity is guaranteed to be executed by a single goroutine.
```

Comment:
The comment is confusing to me. Why `updateCapacity is guaranteed to be executed by a single goroutine` allows processing `podTrackers ` without lock but not processing `assignedTrackers`?

Reply:
`assignedTrackers` is _read_ on the serving path (from `try`). So it is _written_ from a single goroutine but it's ready from hundreds.

Reply:
So we need to lock in order to update `assignedTrackers`.